ring/partition_ring_editor.go

```diff
@@ -38,6 +38,12 @@ func (l *PartitionRingEditor) RemoveMultiPartitionOwner(ctx context.Context, ins
 	})
 }
 
+func (l *PartitionRingEditor) LockPartitionStateChange(ctx context.Context, partitionID int32, locked bool) error {
```

Comment:
[nit] The name `LockPartitionStateChange()` makes me think this is used for locking, but it's actually used to unlock too. I would change it something that doesn't imply the "lock" operation, like `SetPartitionStateChangeLock()` (just an example to give you the idea).